Zend/zend_compile.c

```diff
@@ -2700,6 +2712,52 @@ static void zend_compile_list_assign(
 
 		zend_verify_list_assign_target(var_ast, old_style);
 
+		if (elem_ast->kind == ZEND_AST_UNPACK) {
+			uint32_t opnum_reset, opnum_fetch;
+			znode value_node, reset_node, tmp_node;
+			uint32_t j;
+			
+			zend_emit_op_tmp(&value_node, ZEND_INIT_ARRAY, NULL, NULL);
+			
+			opnum_reset = get_next_op_number();
+			zend_emit_op(&reset_node, ZEND_FE_RESET_R, expr_node, NULL);
+			
+			for (j = 0; j <i; ++j) {
+				opline = zend_emit_op(NULL, ZEND_FE_FETCH_R, &reset_node, NULL);
+				opline->op2_type = IS_VAR;
+	                        opline->op2.var = get_temporary_variable();
+				//save var?
+				//no need to deal with jmp, since those visited indexes are guaranteed to exist.
+			}
```

Comment:
This may skip different elements than were fetched before, if the indexes/order of the array don't match. For example `list($a, ...$b) = [1 => 1, 0 => 0]`. Then `$a = 0`, but this will skip `1` instead.